File: src/02_algorithms/01_sorting/06_heap_sort/implementations/ts/heap_sort.ts
import {
  heap,
  heapify,
} from "../../../../../01_data_structures/06_tree/02_heap/implementations/ts/heap";

export const heapSort = (array: Array<number>) => {
  heap(array);

  for (let i = array.length - 1; i > 0; i--) {
    [array[0], array[i]] = [array[i], array[0]];

    heapify(array, 0, i);
  }
};

File: src/02_algorithms/01_sorting/06_heap_sort/implementations/go/heap_sort.go
package heapSort

import heap "data-structures-algorithms-problems/src/01_data_structures/06_trees/02_heap/implementations/go"

func HeapSort(array []int) {
	heap.Heap(array)

	for i := len(array) - 1; i >= 1; i-- {
		array[0], array[i] = array[i], array[0]

		heap.Heapify(array, 0, i)
	}
}
